src/Illuminate/Bus/Queueable.php

```diff
@@ -95,7 +140,18 @@ public function dispatchNextJobInChain()
     {
         if (! empty($this->chained)) {
             dispatch(tap(unserialize(array_shift($this->chained)), function ($next) {
+                /* @var \Illuminate\Bus\Queueable $next */
+                if (! in_array('Illuminate\Bus\Queueable', class_uses_recursive($next))) {
```

Comment:
Should this check be checking if Illuminate\Contracts\Queue\ShouldQueue is implemented?

Reply:
not sure. should it check both? 
here is the 5.5 current ....

```php

    /**
     * Dispatch the next job on the chain.
     *
     * @return void
     */
    public function dispatchNextJobInChain()
    {
        if (! empty($this->chained)) {
            dispatch(tap(unserialize(array_shift($this->chained)), function ($next) {
                $next->chained = $this->chained;
            }));
        }
    }
```

Reply:
I suppose it could check on the chain method also, so it would not allow adding non-Queueable jobs to the chain.
Originally i added it because I had a test fail without much warning.